traffic_ops/client_tests/configfiles.go

```diff
@@ -0,0 +1,365 @@
+		w.WriteHeader(http.StatusNotImplemented)
+	}
+}
+
+var EDGE_RECORDS_CONFIG = `
+CONFIG proxy.config.admin.user_id STRING ats
+CONFIG proxy.config.body_factory.template_sets_dir STRING /etc/trafficserver/body_factory
+CONFIG proxy.config.config_dir STRING /etc/trafficserver
+CONFIG proxy.config.diags.debug.enabled INT 1
+CONFIG proxy.config.dns.round_robin_nameservers INT 0
+CONFIG proxy.config.exec_thread.autoconfig INT 0
+CONFIG proxy.config.http.cache.required_headers INT 0
+CONFIG proxy.config.http.connect_attempts_timeout INT 10
+CONFIG proxy.config.http.enable_http_stats INT 1
+CONFIG proxy.config.http.insert_response_via_str INT 3
+CONFIG proxy.config.http.parent_proxy.retry_time INT 60
+CONFIG proxy.config.http.parent_proxy_routing_enable INT 1
+CONFIG proxy.config.http.server_ports STRING 80 80:ipv6 443:proto=http:ssl 443:ipv6:proto=http:ssl
+CONFIG proxy.config.http.slow.log.threshold INT 10000
+CONFIG proxy.config.http.transaction_active_timeout_in INT 0
+CONFIG proxy.config.log.logfile_dir STRING /var/log/trafficserver
+CONFIG proxy.config.proxy_name STRING __FULL_HOSTNAME__
+CONFIG proxy.config.reverse_proxy.enabled INT 1
+CONFIG proxy.config.ssl.CA.cert.path STRING /etc/trafficserver/ssl
+CONFIG proxy.config.ssl.client.CA.cert.path STRING /etc/trafficserver/ssl
+CONFIG proxy.config.ssl.client.cert.path STRING /etc/trafficserver/ssl
+CONFIG proxy.config.ssl.client.private_key.path STRING /etc/trafficserver/ssl
+CONFIG proxy.config.ssl.ocsp.enabled INT 1
+CONFIG proxy.config.ssl.server.cert.path STRING /etc/trafficserver/ssl
+CONFIG proxy.config.ssl.server.private_key.path STRING /etc/trafficserver/ssl
+CONFIG proxy.config.ssl.server.ticket_key.filename STRING NULL
+CONFIG proxy.config.url_remap.remap_required INT 0
+`
+
+var MID_RECORDS_CONFIG = `
+CONFIG proxy.config.admin.user_id STRING ats
+CONFIG proxy.config.body_factory.template_sets_dir STRING /etc/trafficserver/body_factory
+CONFIG proxy.config.config_dir STRING /etc/trafficserver
+CONFIG proxy.config.diags.debug.enabled INT 1
+CONFIG proxy.config.dns.round_robin_nameservers INT 0
+CONFIG proxy.config.exec_thread.autoconfig INT 0
+CONFIG proxy.config.http.cache.required_headers INT 0
+CONFIG proxy.config.http.connect_attempts_timeout INT 10
+CONFIG proxy.config.http.enable_http_stats INT 1
+CONFIG proxy.config.http.insert_response_via_str INT 3
+CONFIG proxy.config.http.parent_proxy.retry_time INT 60
+CONFIG proxy.config.http.parent_proxy_routing_enable INT 1
+CONFIG proxy.config.http.server_ports STRING 80 80:ipv6
+CONFIG proxy.config.http.slow.log.threshold INT 10000
+CONFIG proxy.config.http.transaction_active_timeout_in INT 0
+CONFIG proxy.config.log.logfile_dir STRING /var/log/trafficserver
+CONFIG proxy.config.proxy_name STRING __FULL_HOSTNAME__
+CONFIG proxy.config.reverse_proxy.enabled INT 0
+CONFIG proxy.config.url_remap.remap_required INT 0
+`
+
+func edgeRecordsConfig(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Server", SERVER_STRING)
```

Comment:
Theres an awful lot of boilerplate repeated in these Config get functions (and the API functions as well). 

Could the duplication here be reduced?

Reply:
Probably. You know, originally `SERVER_STRING` didn't exist and I just had it as a string literal everywhere :^). The thing is, I've got this `common` function that sets `Content-Type: application/json` and `Server: Traffic Ops/{{VERSION}} (Mock)`. But sometimes you're sending plain text, so you only need the `Server` header, so I found it hard to justify replacing this one line that's everywhere with a different one line that just resolves to a one-line function. 

What'd be great is to put in some middleware, which I'm not really sure how to do. My Go-fu is weak.

Reply:
This would cut way down on size of the code and should hold us for a fairly long time. I don't see ORT sending PUTS/POSTS to the config generation or API routes anytime soon. 

I was thinking something like: 
```
func edgeRecordsConfig(w, r) {
  sendRaw(w, r, []byte(EDGE_PROFILE_HEADER + EDGE_RECORDS_CONFIG))
}

func sendRaw(w, r, contents) {
  w.Header().Set("Server")
  if r.Method == http.MethodGet {
		w.Write(contents)
  } else {
		w.Header().Set("Allow", http.MethodGet)
		w.WriteHeader(http.StatusNotImplemented)
	}
}
```

Reply:
that's a good idea. That should actually probably be `http.StatusMethodNotAllowed`, now that I think about it. Because yeah, nothing should be doing anything to a config file except GET(/HEAD/OPTIONS ?).